multiraft/storage.go

```diff
@@ -233,10 +233,17 @@ func (w *writeTask) start(stopper *stop.Stopper) {
 				groupResp := &groupWriteResponse{raftpb.HardState{}, -1, -1, groupReq.entries}
 				response.groups[groupID] = groupResp
 				if !raft.IsEmptySnap(groupReq.snapshot) {
-					err := group.ApplySnapshot(groupReq.snapshot)
-					if err != nil {
-						panic(err) // TODO(bdarnell)
-					}
+					gID := groupID
+					s.sendEvent(&EventSnapshot{
```

Comment:
I think this event should be sent from `handleWriteResponse` instead of from the `writeTask` goroutine. We can just remove the snapshot from `groupWriteRequest` and `ApplySnapshot` from `WriteableGroupStorage`.